dcc-portal-ui/app/scripts/oncogrid/js/controller.js

```diff
@@ -113,4 +113,119 @@
         };
 
     });
+
+    module.controller('OncoGridUploadController', function(Restangular, SetNameService, donorsLimit, genesLimit, filters, 
+        LocationService, $scope, $modalInstance, $location, $q, SetService, $timeout){
+
+        $scope.donorsLimit = donorsLimit;
+        $scope.genesLimit = genesLimit;
+        $scope.filters = filters;
+        $scope.isLaunchingOncoGrid = false;
+
+        $scope.maxDonorsLimit = 3000;
+        $scope.maxGenesLimit = 100;
+
+        $scope.params = {};
+        $scope.hasValidParams = false;
+
+        $scope.params.donorsCount = Math.min($scope.donorsLimit || $scope.maxDonorsLimit, $scope.maxDonorsLimit);
```

Comment:
This is just initially setting the value of donor count. This parameter `params.donorsCount` is also being used as a `ng-model` in UI. So the value of this parameter will keep changing based on user input. Therefore we can not use it as a base count for validation.